internal/util/osio_token.go

```diff
@@ -0,0 +1,136 @@
+package util
+
+import (
+	"crypto/rsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/satori/go.uuid"
+)
+
+type User struct {
+	Data Data `json:"data"`
+}
+
+type Data struct {
+	Attributes Attributes `json:"attributes"`
+}
+
+type Attributes struct {
+	Username string `json:"username"`
+	Email    string `json:"email"`
+	FullName string `json:"fullName"`
+}
+
+type Result struct {
+	Tokens []TokenResult `json:"tokens"`
+}
+
+type TokenResult struct {
+	UUID  string `json:"uuid"`
+	Token string `json:"token"`
+}
+
+type Token struct {
+	Data Data `json:"data"`
+}
+
+func CreateOSIOToken(env string, uuid string, key string, session string) (string, error) {
+	privateKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(key))
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to parse private key file: %s", err))
+	}
+
+	user, err := loadUser(env, uuid)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to GET user for uuid '%s': %s", uuid, err))
+	}
+
+	token, err := generateToken(env, privateKey, user, uuid, session)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to generate token for uuid '%s': %s", uuid, err))
+	}
+
+	return token, nil
+}
+
+func generateToken(env string, key *rsa.PrivateKey, user User, userID string, session string) (string, error) {
+	token := jwt.New(jwt.SigningMethodRS256)
+	//token.Header["kid"] = "0lL0vXs9YRVqZMowyw8uNLR_yr0iFaozdQk9rzq2OVU"
+	token.Header["kid"] = "PE6-BEECZZpPZIVxLR6NinbthOHJcGqYrfl8v7v6BVA"
```

Comment:
Use the second kid from https://auth.prod-preview.openshift.io/api/token/keys for prod-preview
and the second kid from https://auth.openshift.io/api/token/keys for prod.